src/Frontend/Compiler/JsCompiler.php

```diff
@@ -56,7 +56,7 @@ protected function save(string $file, array $sources): bool
 
         $this->assetsDir->put($file, implode("\n", $output));
 
-        $mapTemp = @tempnam(storage_path('tmp'), $mapFile);
```

Comment:
Any particular reason for switching out the storage dir we include in flarum/flarum for php's default tmp dir? Not opposed, just curious

Reply:
This was the faster solution compared to trying to inject `Paths` - and the sys temp dir does the trick. :shrug:

Reply:
We cannot rely on sys_get_temp_dir() on shared hosting. In addition the explanation for `tempnam` is that it wraps around `sys_get_temp_dir`:

> Creates a file with a unique filename, with access permission set to 0600, in the specified directory. If the directory does not exist or is not writable, tempnam() may generate a file in the system's temporary directory, and return the full path to that file, including its name. 

I think the compiler has to rely on the storage directory. Let's revert this change because it will break shared hosting installations and badly configured environments where a tmp sys dir isn't writeable.

Reply:
I tend to think the key word here is "badly configured environments", not breaking them. :thinking: 

For reference: https://github.com/flarum/core/issues487

Reply:
Potentially also relevant: [my comment here](https://github.com/flarum/core/pull/2117#issuecomment-611979392). Not sure if this is what your comment is about, @luceos. Any sources or pointers would be helpful.

Reply:
> For reference: #487

It seems like the precedent here is using the flarum storage dir? If I understand https://github.com/flarum/core/pull/2117#issuecomment-611979392 correctly and its no longer relevant, I'm fine with this approach.

Reply:
I'm not 100% sure, which is why I was hoping for clarification from @luceos.

Reply:
My comments are based on experience with upload. Using a system tmp directory doesn't always guarantee success.

Reply:
Sorry to be a bit thick - I was hoping you could expose on the "using" part. Above, I linked [to a comment of mine](https://github.com/flarum/core/pull/2117#issuecomment-611979392) that has an example of "using" the system tmp directory that was only problematic due to the specific implementation (checking for access to the *entire directory*, but not files inside). So do we have a detailed explanation somewhere, or a place to link to? :pray: 

I fully believe you there are other problems, but I would love to read about them...

Reply:
I browsed the net for evidence of what I said, but found nothing. The upload repository hasn't held any reference to the function involved. So.. it seems I have been mistaken, sorry for wasting your time with this.

Reply:
@luceos Oh, haha, I was about to say I am ready to change back to storage dir. And now I found this: https://github.com/PHPOffice/PHPWord/issues/310 :laughing:

Reply:
Ooh. Great find! That's exactly what I was looking for 😂

The safest bet is always to work within the confines of the web space, especially in shared hosting environments where isolation matters.

A drawback is that we need to take care of the cleanup during failed uploads.. Something a catch should be able to handle.

Reply:
Oh, look what I found, haha: https://github.com/flarum/core/commit/54660ebd6311f9ea142f1b573263d0d907400786 and #487. We already had all the information here, sorry for the confusion.

Anyway, it turns out the tmp file should not even be necessary, it was just circumventing the fact that the sourcemap library does not have a method for getting the file contents without writing it.

I'll push that fix soon-ish.

Reply:
Pushed it finally. :fireworks: